requirements.txt

```diff
@@ -6,3 +6,4 @@ databroker
 ophyd
 pyzmq
 requests
+json-rpc
```

Comment:
Can we also add `matplotlib` and `pyqt5` here as well? That way it will be installed without installing requirements from `requirements-dev.txt` as these packages are run dependencies (good for future packaging/deployments).

Reply:
I am not sure about including `matplotlib` and `pyqt5` into `requirements.txt`. It will try to install `pyqt5` from PyPI every time one tries to install the package from source. We typically have `pyqt` installed from `nsls2forge`. I am thinking of adding `scikit-image` to the requirements.

Reply:
Actually `bluesky-queueserver` doesn't use `PyQt` or `Matplotlib` (at least not yet). Those should be requirements of Bluesky or they should be installed with Bluesky.

Reply:
These should definitly not depend on mpl and pyqt

Reply:
OK, I just noticed it was missing them during my run a few PRs ago. I'll check later if it's still needed for anything in a fresh environment.

Reply:
It's still hitting me:
```py
(qs2) ✘-INT ~/src/mrakitin/bluesky-queueserver [master|✔] 
10:55 $ start-re-manager
Traceback (most recent call last):
  File "/home/mrakitin/miniconda3/envs/qs2/bin/start-re-manager", line 33, in <module>
    sys.exit(load_entry_point('bluesky-queueserver', 'console_scripts', 'start-re-manager')())
  File "/home/mrakitin/miniconda3/envs/qs2/bin/start-re-manager", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/home/mrakitin/miniconda3/envs/qs2/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 105, in load
    module = import_module(match.group('module'))
  File "/home/mrakitin/miniconda3/envs/qs2/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/mrakitin/src/mrakitin/bluesky-queueserver/bluesky_queueserver/manager/start_manager.py", line 6, in <module>
    from .worker import RunEngineWorker
  File "/home/mrakitin/src/mrakitin/bluesky-queueserver/bluesky_queueserver/manager/worker.py", line 16, in <module>
    from bluesky.callbacks.best_effort import BestEffortCallback
  File "/home/mrakitin/miniconda3/envs/qs2/lib/python3.7/site-packages/bluesky/callbacks/best_effort.py", line 11, in <module>
    import matplotlib.pyplot as plt
ModuleNotFoundError: No module named 'matplotlib'
```

Reply:
MPL is needed to run bluesky-queueserver with Bluesky, so it needs to be installed with Bluesky., but Matplotlib is not the dependency for bluesky-queueserver itself. And actually MPL is not listed as a dependency of Bluesky. The tutorial recommends the following command to install Bluesky and its dependencies:
```
conda install -c nsls2forge bluesky ophyd databroker matplotlib pyqt=5 ipython
```

Reply:
I agree. I wonder why bluesky does not depend on matplotlib then? @tacaswell, what was the reasoning? Maybe we can do it through extra requirements there?

Reply:
I think that the reason is that Bluesky may be used without pyqt and Matplotlib. It depends on callbacks used for data presentation.

Reply:
I agree. Matplotlib should come as a dependency of bluesky (`BestEffortCallback` and friends).

We tested it yesterday, and for the queueserver it was enough to install `matplotlib` alone, without `PyQt5`, and the server was functioning properly. Surprisingly, PyQt5 v5.15 did not play nice with Matplotlib. @tacaswell, have you encountered this issue?